Allow building outside of git repo

If I download a source tarball and try to build it, I quickly run up against this:
```
++ basename cmd/virt-controller
+ BIN_NAME=virt-controller
++ git describe --always --tags
fatal: Not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
+ ARCH_BASENAME=virt-controller-
+ finish
+ docker stop 247a6a54a4574e6be672e1d1312aa76b07bd38df4dfda3a9c5590dffe1025a8f
247a6a54a4574e6be672e1d1312aa76b07bd38df4dfda3a9c5590dffe1025a8f
+ docker rm -f 247a6a54a4574e6be672e1d1312aa76b07bd38df4dfda3a9c5590dffe1025a8f
247a6a54a4574e6be672e1d1312aa76b07bd38df4dfda3a9c5590dffe1025a8f
Makefile:15: recipe for target 'build' failed
make: *** [build] Error 128
```
This is from running **bash -x hack/buiild_go.sh**

As you can see, the script tries to derive the current version from a git tag, but if this isn't in a git repo in the first place, no such thing exists.  Relying on git tags for version information is shaky at best.  I'm not sure of the best way to approach this.  Old-school projects would make version info part of the autoconf info, or have a VERSION file or something like that, but maybe there's a more modern approach that I don't know about.

I'm happy to take this on if the project maintainers show a preference.

--- a/Makefile
+++ b/Makefile
@@ -12,7 +12,7 @@ client-python:
 	hack/dockerized "./hack/generate.sh && TRAVIS_TAG=${TRAVIS_TAG} ./hack/gen-client-python/generate.sh"
 
 build:
-	hack/dockerized "./hack/check.sh && ./hack/build-go.sh install ${WHAT}"
+	hack/dockerized "./hack/check.sh && KUBEVIRT_VERSION=${KUBEVIRT_VERSION} ./hack/build-go.sh install ${WHAT}"
 
 goveralls:
 	SYNC_OUT=false hack/dockerized "./hack/check.sh && TRAVIS_JOB_ID=${TRAVIS_JOB_ID} TRAVIS_PULL_REQUEST=${TRAVIS_PULL_REQUEST} TRAVIS_BRANCH=${TRAVIS_BRANCH} ./hack/goveralls.sh"

--- a/cmd/virt-api/Dockerfile
+++ b/cmd/virt-api/Dockerfile
@@ -25,14 +25,6 @@ RUN useradd -u 1001 --create-home -s /bin/bash virt-api
 WORKDIR /home/virt-api
 USER 1001
 
-# Configure swagger
-RUN curl -OL https://github.com/swagger-api/swagger-ui/tarball/38f74164a7062edb5dc80ef2fdddda24f3f6eb85/swagger-ui.tar.gz \
-    && mkdir swagger-ui && tar xf swagger-ui.tar.gz -C swagger-ui --strip-components 1 \
-    && mkdir third_party \
-    && mv swagger-ui/dist third_party/swagger-ui && rm -rf swagger-ui \
-    && sed -e 's@"http://petstore.swagger.io/v2/swagger.json"@"/swaggerapi/"@' -i  third_party/swagger-ui/index.html \
-    && rm swagger-ui.tar.gz && rm -rf swagger-ui
-
 COPY virt-api /virt-api
 
 ENTRYPOINT [ "/virt-api" ]

--- a/hack/build-go.sh
+++ b/hack/build-go.sh
@@ -71,7 +71,7 @@ for arg in $args; do
     elif [ "${target}" = "install" ]; then
         eval "$(go env)"
         BIN_NAME=$(basename $arg)
-        ARCH_BASENAME=${BIN_NAME}-$(git describe --always --tags)
+        ARCH_BASENAME=${BIN_NAME}-${KUBEVIRT_VERSION}
         mkdir -p ${CMD_OUT_DIR}/${BIN_NAME}
         (
             cd $arg

--- a/hack/common.sh
+++ b/hack/common.sh
@@ -23,3 +23,20 @@ function build_func_tests() {
 PROVIDER=${PROVIDER:-vagrant-kubernetes}
 provider_prefix=${JOB_NAME:-${PROVIDER}}${EXECUTOR_NUMBER}
 job_prefix=${JOB_NAME:-kubevirt}${EXECUTOR_NUMBER}
+
+# Populate an environment variable with the version info needed.
+# It should be used for everything which needs a version when building (not generating)
+# IMPORTANT:
+# RIGHT NOW ONLY RELEVANT FOR BUILDING, GENERATING CODE OUTSIDE OF GIT
+# IS NOT NEEDED NOR RECOMMENDED AT THIS STAGE.
+
+function kubevirt_version() {
+    if [ -n "${KUBEVIRT_VERSION}" ]; then
+        echo ${KUBEVIRT_VERSION}
+    elif [ -d ${KUBEVIRT_DIR}/.git ]; then
+        echo "$(git describe --always --tags)"
+    else
+        echo "undefined"
+    fi
+}
+KUBEVIRT_VERSION="$(kubevirt_version)"

